### ISSUE TITLE
Determine branding for WindowsDesktop shared framework bundle installer

### ISSUE THREAD

**Issue by dagood** [dotnet/core-setup#7530]
This is what the Windows Desktop Runtime bundle installer UI looks like with my initial implementation, basically just replacing `.NET Core` with `Windows Desktop`. Initial questions:

1. Is `Microsoft Windows Desktop Runtime` the right name?
2. What should the description text be changed to?
3. What should/can we do about the overflow? 
   * An interesting wrinkle with fixing it is the `Alpha 1`/`Preview 9`/etc. prerelease label won't appear in the stable release, so if we fix it by e.g. decreasing font size, it would be pointlessly small at 3.0.0 RTM. I'm thinking it might make the most sense to verify it'll eventually work with RTM versioning and leave it.

![image](https://user-images.githubusercontent.com/12819531/62241193-5132d000-b39e-11e9-967e-d88bbbde1ba7.png)

Note that this has to loop through the loc process once we're done. I'll drive that.

@vatsan-madhavan @merriemcgaw @zsd4yr, thoughts?

/cc @dleeapho @leecow @ericstj

(Bundle implementation tracked at https://github.com/dotnet/core-setup/issues/6384.)

**Comment by vatsan-madhavan** [comment]
My 2c - Let's stick to a name that matches what we already selected for the runtime & sdk etc. -  _Windows Desktop_. In other words, I'm happy with @dagood's proposal for _name_ - **Windows Desktop Runtime**

**Comment by grubioe** [comment]
+ @BethMassi for input. 

Windows Desktop Runtime works from my perspective.

**Comment by merriemcgaw** [comment]
I'm fine with that or with "Windows Desktop Shared Framework Runtime", but that seems even more excessive :)

**Comment by dagood** [comment]
Cool, sounds like we agree on this so far, which perfectly mirrors what shows up in the .NET Core Runtime installer:

* Title: **Microsoft Windows Desktop Runtime - 5.0.0 Alpha 1 (x64)**
* Description header: **Windows Desktop Runtime**

I believe branding is consistent across the product with calling the shared frameworks "runtimes", so no need to add the extra words. 🙂 (In some contexts I might call CoreCLR or Mono a runtime, making things a bit confusing... I'm trying to call them VMs instead when that comes up.)

---

Any ideas on the description body and links? Is there someone I can assign this issue to to figure that out?

**Comment by dagood** [comment]
Current text:

> ### Windows Desktop Runtime
> 
> .NET Core is a development platform that you can use to build command-line applications, microservices and modern websites. It is open source, cross-platform, and supported by Microsoft. We hope you enjoy it!

My motivation for changing this:

1. The text describes development and building, which this isn't for.
   * Depending on our guidance, developers may point end users at this installer to be able to launch their FDE/FDD apps. (https://github.com/dotnet/core-setup/issues/6381, https://github.com/dotnet/core/issues/2618.)
2. It mentions various types of projects, but desktop apps are notably missing.
3. It says `cross-platform`, but it don't run/exist on non-Windows platforms.

Maybe for Preview 9, we can simply remove the section. I'll submit a quick PR to do this as the default.

**Comment by vatsan-madhavan** [comment]
This is what I see for ASP.NET Core. Are we planning on changing this or adding more verbiage to it? If the answer is **no**, then I'd recommend that we take out the text from WindowsDesktop shared framework as well and keep things consistent. 

![image](https://user-images.githubusercontent.com/20246435/62669181-8c577500-b943-11e9-824e-edbf102de81f.png)

If we plan on adding some text to ASP.NET Core shared framework installer, then we should definitely make a matching update and add relevant text to WindowsDesktop shared framework as well. Something like this could be a reasonable starting point for iteration: 

> Windows Desktop shared framework/runtime contains support for using Windows Presentation Foundation (WPF) and Windows Forms.  It is a part of .NET Core development platform, is open-source, and supported by Microsoft.

**Comment by merriemcgaw** [comment]
@vatsan-madhavan  - Why WPF first:
>Windows Desktop shared framework/runtime contains support for using Windows Presentation >Foundation (WPF) and Windows Forms. It is a part of .NET Core development platform, is open->source, and supported by Microsoft.

😁

I actually really like just keeping it simple like the current text for ASP.NET. Either that or we arm-wrestle over who is first 😉

**Comment by vatsan-madhavan** [comment]
@merriemcgaw, No idea - I copied it from https://docs.microsoft.com/en-us/dotnet/core/whats-new/dotnet-core-3-0#windows-desktop ;-). I’m fine with any order really - let’s change it up by all means!

**Comment by dagood** [comment]
That installer's text is a little busted (note the double space and weird phrase), here's the 2.2 one:

![image](https://user-images.githubusercontent.com/12819531/62715176-95892600-b9c5-11e9-9213-0c7ab0d020fd.png)

Might want to report the 3.0 issue (if it hasn't been already)?

The audience seems to me to be different enough that I don't know if it makes sense to take cues from that installer. (Devs installing the sfx on server/hosting machines vs. users (and devs) trying to run a FDE/FDD WindowsDesktop app.) But the SDK and .NET Core Runtime installers are also dev-focused, and they're the wordy ones, so I don't see clear precedent either way.

**Comment by merriemcgaw** [comment]
@vatsan-madhavan maybe we can alternate the order each major release to keep it fair ;)

**Comment by dagood** [comment]
I'm going to move a little forward with porting the installer to 3.0 with the one-liner and description removed--this means we can reuse the .NET Core Runtime loc, because those lines appear to be the only text that wouldn't work for both.

![image](https://user-images.githubusercontent.com/12819531/63188549-4a0af380-c027-11e9-8446-230f4cea7134.png)

If there's text that should be there for Preview 9, I think we can add it, but I doubt it could be localized in time.

**Comment by richlander** [comment]
Proposal:

- Microsoft .NET Core for Windows Desktop Apps -- 3.0.0 (x64)
- Microsoft .NET Core for Windows Server Apps -- 3.0.0
- Microsoft ASP.NET Core Runtime -- 3.0.0
- Microsoft .NET Core Runtime -- 3.0.0 (x64)

Note: The first two are intended for direct customer download and usage since they are for specific app-models. The SDK is similar in that respect. The bottom two are not useful in most scenarios and are largely pulled in via other installers.

Currently (for comparison):

- Microsoft Windows Desktop Runtime -- 3.0.0 (x64)
- Microsoft .NET Core 3.0.0 Windows Server Hosting
- Microsoft ASP.NET Core 3.0.0 Shared Framework
- Microsoft .NET Core Runtime -- 3.0.0 (x64)

**Comment by BethMassi** [comment]
Agree with Rich on naming. Branding still falls under .NET Core.

**Comment by vatsan-madhavan** [comment]
I like @richlander’s proposal! 

Yes, WindowsDesktop should include both x86 and x64. @dagood can you confirm ?

**Comment by dagood** [comment]
The download page https://dotnet.microsoft.com/download/dotnet-core/3.0 has this currently:

> Windows
> * ASP.NET Core/.NET Core: [Runtime & Hosting Bundle](#)
> * ASP.NET Core Installer: [x64](#) | [x86](#)
> * ASP.NET Core Binaries: [x64](#) | [x86](#) | [ARM32](#)
> * .NET Core Installer: [x64](#) | [x86](#)
> * .NET Core Binaries: [x64](#) | [x86](#) | [ARM32](#)

I believe only the `Runtime & Hosting Bundle` includes both x64 and x86. Another important difference is that the `Runtime & Hosting Bundle` includes the .NET Core Runtime. Others have one installer per arch and only includes a single shared framework.

Here's what a "complete" lineup looks like to me, based on the wording on the download page plus the proposal:

* ASP.NET Core/.NET Core: [Runtime & Hosting Bundle](#)
* Windows Desktop/.NET Core: [Runtime & Windows Desktop Bundle](#)
* ASP.NET Core Installer: [x64](#) | [x86](#)
* ASP.NET Core Binaries: [x64](#) | [x86](#) | [ARM32](#)
* Windows Desktop Installer: [x64](#) | [x86](#)
* Windows Desktop Binaries: [x64](#) | [x86](#)
* .NET Core Installer: [x64](#) | [x86](#)
* .NET Core Binaries: [x64](#) | [x86](#) | [ARM32](#)

This gets a bit muddy because the branding uses "Bundle" to mean that it includes the .NET Core Runtime, but all five of those are "bundle installers" in technical language. This is why issue titles might be a little confusing.

The installer that I've implemented is `Windows Desktop Installer`, one installer for each arch. I haven't had the `Runtime & Windows Desktop Bundle` in mind, but it makes sense that it's needed for app users. Filed https://github.com/dotnet/core-setup/issues/7763 to track it.

I don't have time to create a `Runtime & Windows Desktop Bundle` in 3.0 Preview 9. The infra builds x64 and x86 in parallel, so there is some work to do to even get the assets from both archs on the same machine to combine them together.

Continuing to move forward with the `Windows Desktop Installer` that has no description text, for now.

**Comment by DamianEdwards** [comment]
I like @richlander's proposal above. That gives us something that I think will translate to what we need to display on the main [download](https://dotnet.microsoft.com/download) page too (basically we'd add a button to download the desktop apps runtime).

Richard also suggested we combine both architectures into the installer for Windows Desktop app runtimes. This would simplify the download choices but of course increase the installer size. Does anyone know what the .NET Framework installer does RE architectures today?

**Comment by dleeapho** [comment]
> Does anyone know what the .NET Framework installer does RE architectures today?

@NikolaMilosavljevic can you answer?

**Comment by vatsan-madhavan** [comment]
.NET Framework follows Window’s rules. 

Installing x64 will (in fact, **must**) implicitly install WoW64 binaries (ie x86). 

There is no such thing as registration-free installation (ie, “zip”, “xcopy” etc) - an “installer” is always involved, so managing WoW64 gets handled by the installer (MSI, CBS etc).

**Comment by dagood** [comment]
For tracking purposes, this issue is about finding the right branding for the installers we're producing for 3.0 that only include the Windows Desktop runtime and only include a single architecture. (Windows Desktop Installer: x64 | x86 above.)

The installer that bundles x86 + x64 + WindowsDesktop + .NET Core is tracked at https://github.com/dotnet/core-setup/issues/7763 and it's unclear whether it's necessary with other plans for 3.1.

**Comment by dagood** [comment]
Moving to dotnet/windowsdesktop. If there are changes based on this issue to backport to 3.1, please open a new issue in dotnet/core-setup.
